File: matiasmasso/Mat.NET/DT2/DTOCliProductBlocked.vb
Public Class DTOCliProductBlocked
    Property Contact As DTOContact = Nothing
    Property Product As DTOProduct
    Property DistModel As DistModels
    Property Cod As Codis
    Property Zip As String
    Property Obs As String

    Property LastFch As Date
    Property IsNew As Boolean
    Property IsLoaded As Boolean

    Public Enum DistModels
        NotSet
        Open 'lliure
        Closed 'llista tancada de distribuidors oficials
    End Enum

    Public Enum Codis
        Standard
        Exclusiva
        NoAplicable
        Exclos
        DistribuidorOficial
        AltresEnExclusiva
    End Enum

    Shared Function Factory(oContact As DTOContact, Optional oProduct As DTOProduct = Nothing) As DTOCliProductBlocked
        Dim retval As New DTOCliProductBlocked
        With retval
            .Contact = oContact
            .Product = oProduct
        End With
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOCliProductBlocked.cs
using Newtonsoft.Json.Linq;
using System;

namespace DTO
{
    public class DTOCliProductBlocked
    {
        public DTOContact contact { get; set; } = null/* TODO Change to default(_) if this is not a reference type */;
        public Object product { get; set; }
        public DistModels distModel { get; set; }
        public Codis cod { get; set; }
        public string zip { get; set; }
        public string obs { get; set; }

        public DateTime lastFch { get; set; }
        public bool IsNew { get; set; }
        public bool IsLoaded { get; set; }

        public enum DistModels
        {
            notSet,
            open // lliure
    ,
            closed // llista tancada de distribuidors oficials
        }

        public enum Codis
        {
            standard,
            exclusiva,
            noAplicable,
            exclos,
            distribuidorOficial,
            altresEnExclusiva
        }

        public static DTOCliProductBlocked Factory(DTOContact oContact, DTOProduct oProduct = null/* TODO Change to default(_) if this is not a reference type */)
        {
            DTOCliProductBlocked retval = new DTOCliProductBlocked();
            {
                var withBlock = retval;
                withBlock.contact = oContact;
                withBlock.product = oProduct;
            }
            return retval;
        }

        public void RestoreObjects()
        {
            this.product = DTOProduct.fromJObject(this.product as JObject);
        }
    }
}
